keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/BooleanProxyVertex.java

```diff
@@ -28,12 +29,25 @@ public BooleanProxyVertex(VertexLabel label) {
         this(Tensor.SCALAR_SHAPE, label);
     }
 
+    @ExportVertexToPythonBindings
     public BooleanProxyVertex(long[] shape, VertexLabel label) {
         super(shape);
-        this.setLabel(label);
+        setLabel(label);
+    }
+
+    @Override
+    public void addScopeToLabel(String scope) {
+        super.setLabel(this.getLabel().withExtraNamespace(scope));
+    }
+
+    @Override
+    public <V extends Vertex<BooleanTensor>> V setLabel(VertexLabel label) {
```

Comment:
I'm not sure that makes sense for a boolean vertex to extend a double vertex. We could put this as a default method in the ProxyVertex interface but we would then have to make sure all ProxyVertices _extend_ Vertex (so they can override .setLabel) which is conceptually different to the current interface for ProxyVertices.

Reply:
Ah, my bad: ** `Vertex`

Reply:
Ah I see. It still stands that it would be a bit of a conceptual change from the current implementation which goes more down the 'Composition over inheritance' route.